### PR TITLE
Update to FAPI `0.91.0+1.20.1`

### PR DESCRIPTION
**Pull request by Kneelawk**
# This PR
This PR updates QFAPI to match FAPI `0.91.0+1.20.1`.

# Associated
* While making this update, an issue in QSL was discovered: https://github.com/QuiltMC/quilt-standard-libraries/issues/357

### CODE REVIEW

**Comment by Kneelawk** [comment]
I can squash some of the commits if need be.

**Review by eristhea — APPROVED** [review]

**Comment by eristhea** [comment]
> I can squash some of the commits if need be.

we don't squash commits on QFAPI, don't worry about it

**Comment by OroArmor** [comment]
What is the status on this PR? does it need to sync QSL?

**Comment by OroArmor** [comment]
Edit: Oh i see. no QSL sync needed. How is the rest of the PR?

File: gradle.properties [hunk 1]
```diff
@@ -4,7 +4,7 @@ fabric.loom.multiProjectOptimisation=true
 
 archivesBaseName=quilted_fabric_api
 version=7.4.0
```
Comment (OroArmor) [inline gradle.properties]:
```suggestion
version=7.5.0
```

**Review by OroArmor — APPROVED** [review]

**Pull request merged**